File: backend/src/main/java/com/scinar/giderlerim/service/AuthService.java
package com.scinar.giderlerim.service;

import com.scinar.giderlerim.dto.request.GirisRequest;
import com.scinar.giderlerim.dto.request.KayitRequest;
import com.scinar.giderlerim.dto.response.ApiResponse;
import com.scinar.giderlerim.dto.response.TokenResponse;

public interface AuthService {
    ApiResponse<TokenResponse> kayitOl(KayitRequest request);
    ApiResponse<TokenResponse> girisYap(GirisRequest request);
    ApiResponse<TokenResponse> tokenYenile(String refreshToken);
}

File: frontend/src/services/authService.ts
import apiClient from './apiClient';
import { ApiResponse } from '@/types/api.types';
import { GirisRequest, KayitRequest, TokenResponse } from '@/types/auth.types';

export const authService = {
  async kayitOl(data: KayitRequest): Promise<ApiResponse<TokenResponse>> {
    const response = await apiClient.post<ApiResponse<TokenResponse>>('/auth/kayit', data);
    return response.data;
  },

  async girisYap(data: GirisRequest): Promise<ApiResponse<TokenResponse>> {
    const response = await apiClient.post<ApiResponse<TokenResponse>>('/auth/giris', data);
    return response.data;
  },

  async tokenYenile(refreshToken: string): Promise<ApiResponse<TokenResponse>> {
    const response = await apiClient.post<ApiResponse<TokenResponse>>(
      '/auth/token-yenile',
      null,
      { headers: { 'X-Refresh-Token': refreshToken } }
    );
    return response.data;
  },
};
